src/main/java/org/myrobotlab/service/Servo.java

```diff
@@ -836,34 +835,47 @@ public double getPos() {
    * 
    * https://www.google.com/search?q=attach+myrobotlab&oq=attach+myrobotlab
    */
-  public void attachServoController(ServoController controller) throws Exception {
-    if (!(controller == null)) {
-      if (isAttachedServoController(controller)) {
-        log.info("{} servo is already attached to controller {}", getName(), this.controller.getName());
-        return;
-      } else if (this.controller != null && this.controller != controller) {
-        // we're switching controllers - need to detach first
-        detach();
-      }
+  public void attachServoController(Attachable attachable) throws Exception {
+    
+    ServoController sc = null;
+    if (attachable == null){
+      return;
+    }
+    
+    // find servo controller - 
+    if (attachable instanceof ServoController) {
+      sc = (ServoController)attachable;
+    } else if (attachable instanceof Simulator) {
+      sc = ((Simulator)attachable).getServoController();
+    } else {
+      log.error("{} unsupported type {}", attachable.getName(), Attachable.class.getCanonicalName());

```

Comment:
A log message here is fine, however, this method signature says it throws an exception.  I think if you pass in a bogus service, you should throw something here.. not just log it and silently return..

Reply:
I don't mind either way ... you want it to go BOOM bigger .. fine by me